src/libs/ajax/DatasetBuilder.ts

```diff
@@ -12,33 +12,47 @@ import {
 } from 'src/libs/ajax/DataRepo';
 
 /** A specific criteria based on a type. */
-export interface Criteria {
+export interface Criteria extends CriteriaApi {
+  count?: number;
+}
+export interface CriteriaApi {
   kind: 'domain' | 'range' | 'list';
   name: string;
   id: number;
-  count?: number;
 }
-
-export interface DomainCriteria extends Criteria {
+export interface DomainCriteriaApi extends Criteria {
   kind: 'domain';
+}
+
+export interface DomainCriteria extends DomainCriteriaApi {
   domainOption: DomainOption;
 }
 
-export interface ProgramDataRangeCriteria extends Criteria {
+export interface ProgramDataRangeCriteriaApi extends Criteria {
```

Comment:
Is this correct? I would expect
```suggestion
export interface ProgramDataRangeCriteriaApi extends CriteriaApi {
```